### PR TITLE
feat(release): add windows msi installers

### PR DESCRIPTION
**Pull request by davidkna**
#### Description

This PR adds the required configurations files and changes to the release workflow to publish MSI installers. Windows aarch64 builds don't get installers yet, because they're not yet supported by wix.
As there's no easy way to include MSVC-redist in the installer, this PR also changes the Windows builds to use static linking. This is the approach recommended by cargo-wix.

#### Motivation and Context

Required for #1295
Related #690

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -93,13 +98,28 @@ jobs:
           profile: minimal
           target: ${{ matrix.target }}
 
+      - name: Setup | Install cargo-wix [Windows]
```
Comment (matchai) [inline .github/workflows/release.yml:101]: A passing thought — not something we have to address in this PR:
Our release-time build process is getting pretty complex with building installers and notarizing, which leads to a lot of unexpected failures at the time of release.

For this, in particular, I think there'd be value in adding this to CI. Maybe not on every commit, like the rest of our suite, but perhaps on every push to `master`.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -93,13 +98,28 @@ jobs:
...
+        # aarch64 is only supported in wix 4.0 development builds
+        if: matrix.os == 'windows-latest' && matrix.target != 'aarch64-pc-windows-msvc'
+        run: cargo install --version 0.3.2 cargo-wix
```
Comment (matchai) [inline .github/workflows/release.yml:104]: It's unfortunate that this, being a hard-coded version, won't be updated by Dependabot.
I wish we could have cargo subcommands be retrieved from our dev-dependencies. That's not possible though, right?

**Review by matchai — APPROVED** [review]
It looks to me like `icon.ico` is translucent. Is that intentional?

Left a couple of comments, but nothing blocking. LGTM! 👍

**Comment by davidkna** [comment]
> It looks to me like `icon.ico` is transparent. Is that intentional?

`rsvg-convert` and `inkscape` keep giving me transparent images for some reason. I'll need to investigate a bit more.

**Comment by matchai** [comment]
@davidkna It looks like the icon svg in the repo is transparent. That's probably my fault 😅 
Let me fix that up.

**Comment by matchai** [comment]
@davidkna Done! Feel free to try using the updated `icon.svg`

File: .github/workflows/release.yml (hunk 1 above)
Comment (davidkna) [reply .github/workflows/release.yml:101]: Done.

File: .github/workflows/release.yml (hunk 2 above)
Comment (davidkna) [reply .github/workflows/release.yml:105]: I don't think that's possible.

**Review by matchai — APPROVED** [review]

**Pull request merged**